### PR TITLE
provider/openstack: upgrade asg operations to use lbaasv2

### PR DESCRIPTION
**Pull request by drmaas**

### CODE REVIEW

File: clouddriver-openstack/src/main/groovy/com/netflix/spinnaker/clouddriver/openstack/deploy/description/servergroup/ServerGroupParameters.groovy [hunk 1]
```diff
@@ -95,6 +93,24 @@ class ServerGroupParameters {
   }
 
   /**
+   * Stack parameters of type 'comma_delimited_list' come back as a unicode json string. We need to split that up.
+   *
+   * I need a shower.
```
Comment (varikin) [inline clouddriver-openstack/src/main/groovy/com/netflix/spinnaker/clouddriver/openstack/deploy/description/servergroup/ServerGroupParameters.groovy]: May not want to leave this comment in

File: clouddriver-openstack/src/main/groovy/com/netflix/spinnaker/clouddriver/openstack/deploy/description/servergroup/MemberData.groovy [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.netflix.spinnaker.clouddriver.openstack.deploy.description.servergroup
+
+import groovy.transform.Canonical
+
```
Comment (derekolk) [inline clouddriver-openstack/src/main/groovy/com/netflix/spinnaker/clouddriver/openstack/deploy/description/servergroup/MemberData.groovy:20]: Can you please provide context?  Why and what is it used for?

File: clouddriver-openstack/src/main/groovy/com/netflix/spinnaker/clouddriver/openstack/deploy/ops/servergroup/DeployOpenstackAtomicOperation.groovy [hunk 3]
```diff
@@ -112,28 +107,37 @@ class DeployOpenstackAtomicOperation implements AtomicOperation<DeploymentResult
...
-      task.updateStatus BASE_PHASE, "Finished loading templates"
-
-      task.updateStatus BASE_PHASE, "Getting load balancer details for pool id $description.serverGroupParameters.poolId"
-      LbPool pool = provider.getLoadBalancerPool(description.region, description.serverGroupParameters.poolId)
-      task.updateStatus BASE_PHASE, "Found load balancer details for pool id $description.serverGroupParameters.poolId with name $pool.name"
-
-      task.updateStatus BASE_PHASE, "Getting internal port used for load balancer $pool.name"
-      int port = provider.getInternalLoadBalancerPort(pool)
-      task.updateStatus BASE_PHASE, "Found internal port $port used for load balancer $pool.name"
+      //look up all load balancer listeners -> pool ids and internal ports
```
Comment (derekolk) [inline clouddriver-openstack/src/main/groovy/com/netflix/spinnaker/clouddriver/openstack/deploy/ops/servergroup/DeployOpenstackAtomicOperation.groovy]: Can this be functionality be pulled out into a common method shared across upsertServerGroup/resizeServerGroup/upsertLoadBalancer?

File: clouddriver-openstack/src/main/groovy/com/netflix/spinnaker/clouddriver/openstack/deploy/ops/servergroup/ServerGroupConstants.groovy [hunk 4]
```diff
@@ -0,0 +1,39 @@
...
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.netflix.spinnaker.clouddriver.openstack.deploy.ops.servergroup
+
+/**
```
Comment (derekolk) [inline clouddriver-openstack/src/main/groovy/com/netflix/spinnaker/clouddriver/openstack/deploy/ops/servergroup/ServerGroupConstants.groovy]: Remove empty comment or provide a definition.

File: clouddriver-openstack/src/main/groovy/com/netflix/spinnaker/clouddriver/openstack/deploy/ops/servergroup/ServerGroupConstants.groovy [hunk 5]
```diff
@@ -0,0 +1,39 @@
...
+
+  //this is the name of the subtemplate referenced by the template,
+  //and needs to be loaded into memory as a String
+  final static String SUBTEMPLATE_FILE = 'asg_resource.yaml'
+
+  //this is the name of the member template referenced by the subtemplate,
+  //and is contructed on the fly
+  final static String MEMBERTEMPLATE_FILE = 'asg_member.yaml'
+
+  final static String SUBTEMPLATE_OUTPUT = 'asg_resource'
```
Comment (derekolk) [inline clouddriver-openstack/src/main/groovy/com/netflix/spinnaker/clouddriver/openstack/deploy/ops/servergroup/ServerGroupConstants.groovy]: If the output and files are dependent on each other, then it makes sense to use the OUTPUT with the FILE definition.

**Pull request closed without merging**